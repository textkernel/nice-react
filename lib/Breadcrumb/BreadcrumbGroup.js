import React from 'react';
import PropTypes from 'prop-types';
import props from '../props';

const BreadcrumbGroup = ({ size, children }) => (
  <ol className={`nice-breadcrumb ${size}`}>
    {children}
  </ol>
);

BreadcrumbGroup.propTypes = {
  children: PropTypes.node,
  size: PropTypes.oneOf(props.breadcrumbsSizes)
};

BreadcrumbGroup.defaultProps = {
  children: null,
  size: null
};

export default BreadcrumbGroup;