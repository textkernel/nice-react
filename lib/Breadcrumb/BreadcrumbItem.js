import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class BreadcrumbItem extends Component {

  static propTypes = {
    children: PropTypes.node,
    onClick: PropTypes.func,
    icon: PropTypes.string,
    text: PropTypes.string.isRequired,
    value: PropTypes.string,
    current: PropTypes.bool,
  }

  static defaultProps = {
    children: null,
    onClick: () => true,
    icon: null,
    value: null,
    current: false
  }

  handleClick = (e) => {
    e.preventDefault();
    
    const { onClick, value } = this.props;
    onClick(value);
  }

  render = () => {
    const { children, text, icon, current } = this.props;

    const classes = [];
    if (icon) classes.push(`icon ${icon}`);
    if (current) classes.push('current');

    return (
      <li>
        <a 
          href='javascript:;'
          className={classes.join(' ')}
          onClick={this.handleClick}
        >
          {text}
        </a>

        {
          children &&
            <ul> {children} </ul>
        } 
      </li>
    );
  }

}