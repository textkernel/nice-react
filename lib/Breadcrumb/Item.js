import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Item extends Component {

    handleClick(e) {
        e.preventDefault();
        
        const { onClick, value } = this.props;
        onClick(value);
    }

    render() {
        const { children, text, icon, current } = this.props;

        const classes = [];
        if (icon) classes.push(`icon ${icon}`);
        if (current) classes.push('current');

        return (
          <li>
            <a 
              href="#"
              className={ classes.join(' ') }
              onClick={ this.handleClick }
            >
              {text}
            </a>

            {
              children &&
                <ul> {children} </ul>
            } 
          </li>
        );
    }
}

Item.propTypes = {
    children: PropTypes.node,
    onClick: PropTypes.func,
    icon: PropTypes.string,
    text: PropTypes.string.isRequired,
    value: PropTypes.string,
    current: PropTypes.bool,
};

Item.defaultProps = {
    children: null,
    onClick: () => true,
    icon: null,
    value: null,
    current: false
};

export default Item;
