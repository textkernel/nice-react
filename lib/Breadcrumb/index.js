export { default as BreadcrumbGroup } from './BreadcrumbGroup';
export { default as BreadcrumbItem } from './BreadcrumbItem';