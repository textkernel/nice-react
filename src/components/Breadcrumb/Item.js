import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Item extends Component {

    handleClick = (e) => {
        e.preventDefault();
        
        const { onClick, value } = this.props;
        onClick(value);
    }

    render() {
        const { children, text, icon, current, className, style } = this.props;

        const classes = [className];
        if (icon) classes.push(`icon ${icon}`);
        if (current) classes.push('current');

        return (
          <li>
            <a 
              href="#"
              className={ classes.join(' ') }
              onClick={ this.handleClick }
              style={style}
            >
              {text}
            </a>

            {
              children &&
                <ul> {children} </ul>
            } 
          </li>
        );
    }
}

Item.propTypes = {
    className: PropTypes.string,
    children: PropTypes.node,
    current: PropTypes.bool,
    icon: PropTypes.string,
    onClick: PropTypes.func,
    style: PropTypes.object,
    text: PropTypes.string.isRequired,
    value: PropTypes.string,
};

Item.defaultProps = {
    className: '',
    children: null,
    current: false,
    icon: null,
    onClick: () => true,
    style: {},
    value: null,
};

export default Item;
