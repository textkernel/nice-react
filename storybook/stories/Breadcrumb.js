import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';

import { Breadcrumb } from 'nice-react';

const stories = storiesOf('Breadcrumb', module);
stories.addDecorator((story, context) => withInfo('')(story)(context));
stories.addDecorator(withKnobs);

stories.add('with items', () => (
    <Breadcrumb>
        <Breadcrumb.Item text='Level 1' />
        <Breadcrumb.Item text='Level 2' />
        <Breadcrumb.Item text='Level 3' />
    </Breadcrumb>
));

stories.add('with icons', () => (
    <Breadcrumb>
        <Breadcrumb.Item icon='fa-home'  text='Level 1' />
        <Breadcrumb.Item icon='fa-star'  text='Level 2' />
        <Breadcrumb.Item icon='fa-truck' text='Level 3' />
    </Breadcrumb>
));

stories.add('with one item selected as current', () => (
    <Breadcrumb>
        <Breadcrumb.Item icon='fa-home'  text='Level 1' />
        <Breadcrumb.Item text='Level 2' />
        <Breadcrumb.Item text='Level 3' current />
    </Breadcrumb>
));

stories.add('with child nodes', () => (
    <Breadcrumb>
        <Breadcrumb.Item icon='fa-home'  text='Level 1' />
        <Breadcrumb.Item text='Level 2'>
            <Breadcrumb.Item text='Some child node 1' />
            <Breadcrumb.Item text='Some child node 2' />
        </Breadcrumb.Item>
        <Breadcrumb.Item text='Level 3' />
    </Breadcrumb>
));

stories.add('with onClick', () => (
    <Breadcrumb>
        <Breadcrumb.Item text='Level 1' value='level-1' onClick={value => alert(value)} />
        <Breadcrumb.Item text='Level 2' value='level-2' onClick={value => alert(value)} />
        <Breadcrumb.Item text='Level 3' value='level-3' onClick={value => alert(value)} />
    </Breadcrumb>
));
