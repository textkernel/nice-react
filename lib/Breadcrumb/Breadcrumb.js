import React from 'react';
import PropTypes from 'prop-types';
import props from '../props';
import Item from './Item';

const Breadcrumb = ({ size, children }) => (
  <ol className={`nice-breadcrumb ${size}`}>
    {children}
  </ol>
);

Breadcrumb.Item = Item;

Breadcrumb.propTypes = {
    children: PropTypes.node,
    size: PropTypes.oneOf(props.breadcrumbsSizes)
};

Breadcrumb.defaultProps = {
    children: null,
    size: null
};

export default Breadcrumb;