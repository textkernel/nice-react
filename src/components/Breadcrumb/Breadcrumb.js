import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import props from '../../props';
import Item from './Item';
import './Breadcrumb.scss';

const Breadcrumb = ({ size, children, className, style }) => (
  <ol className={classnames('nice-breadcrumb', size, className)} style={style}>
    {children}
  </ol>
);

Breadcrumb.Item = Item;

Breadcrumb.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    size: PropTypes.oneOf(props.breadcrumbsSizes),
    style: PropTypes.object,
};

Breadcrumb.defaultProps = {
    children: null,
    className: '',
    size: null,
    style: {}
};

export default Breadcrumb;