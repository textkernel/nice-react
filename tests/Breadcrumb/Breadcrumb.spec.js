import React from 'react';
import Breadcrumb from '../../lib/Breadcrumb';

describe('<Breadcrumb />', () => {
    it('should render default Breadcrumb', () => {
        const wrapper = shallow(
          <Breadcrumb>
            <Breadcrumb.Item text='Level 1' />
            <Breadcrumb.Item text='Level 2' />
            <Breadcrumb.Item text='Level 3' />
          </Breadcrumb>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render small Breadcrumb', () => {
        const wrapper = shallow(
          <Breadcrumb size='small'>
            <Breadcrumb.Item text='Level 1' />
            <Breadcrumb.Item text='Level 2' />
            <Breadcrumb.Item text='Level 3' />
          </Breadcrumb>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
