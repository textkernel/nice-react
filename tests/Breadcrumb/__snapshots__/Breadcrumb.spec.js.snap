// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> should render default Breadcrumb 1`] = `
<ol
  className="nice-breadcrumb null"
>
  <Item
    current={false}
    icon={null}
    onClick={[Function]}
    text="Level 1"
    value={null}
  />
  <Item
    current={false}
    icon={null}
    onClick={[Function]}
    text="Level 2"
    value={null}
  />
  <Item
    current={false}
    icon={null}
    onClick={[Function]}
    text="Level 3"
    value={null}
  />
</ol>
`;

exports[`<Breadcrumb /> should render small Breadcrumb 1`] = `
<ol
  className="nice-breadcrumb small"
>
  <Item
    current={false}
    icon={null}
    onClick={[Function]}
    text="Level 1"
    value={null}
  />
  <Item
    current={false}
    icon={null}
    onClick={[Function]}
    text="Level 2"
    value={null}
  />
  <Item
    current={false}
    icon={null}
    onClick={[Function]}
    text="Level 3"
    value={null}
  />
</ol>
`;
